src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/AbstractQueryAndScanMapper.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package com.salesforce.dynamodbv2.mt.mappers.sharedtable.impl;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.salesforce.dynamodbv2.mt.mappers.sharedtable.impl.MappingUtils.getNextFieldPlaceholder;
+import static com.salesforce.dynamodbv2.mt.mappers.sharedtable.impl.MappingUtils.getNextValuePlaceholder;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.amazonaws.services.dynamodbv2.model.ScanRequest;
+import com.amazonaws.services.dynamodbv2.model.ScanResult;
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.salesforce.dynamodbv2.mt.mappers.metadata.DynamoTableDescription;
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.impl.RequestWrapper.AbstractRequestWrapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+abstract class AbstractQueryAndScanMapper implements QueryAndScanMapper {
+
+    protected final DynamoTableDescription virtualTable;
+    private final Function<String, RequestIndex> indexLookup;
+    private final ConditionMapper conditionMapper;
+    private final ItemMapper itemMapper;
+
+    AbstractQueryAndScanMapper(DynamoTableDescription virtualTable,
+                               Function<String, RequestIndex> indexLookup,
+                               ConditionMapper conditionMapper,
+                               ItemMapper itemMapper) {
+        this.virtualTable = virtualTable;
+        this.indexLookup = indexLookup;
+        this.conditionMapper = conditionMapper;
+        this.itemMapper = itemMapper;
+    }
+
```

Comment:
A lot of the logic here was pulled out from RandomPartitioningQueryAndScanMapper, e.g., converting legacy request parameters into key condition / filter expressions.